processor/processor.go

```diff
@@ -1820,9 +1820,16 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob) *transf
 				&backendconfig.DestinationT{},
 			)
 
-			// TODO: TP ID preference 1.event.context set by rudderTyper   2.From WorkSpaceConfig (currently being used)
-			shallowEventCopy.Metadata.TrackingPlanId = source.DgSourceTrackingPlanConfig.TrackingPlan.Id
-			shallowEventCopy.Metadata.TrackingPlanVersion = source.DgSourceTrackingPlanConfig.TrackingPlan.Version
+			trackingPlanId := source.DgSourceTrackingPlanConfig.TrackingPlan.Id
+			trackingPlanVersion := source.DgSourceTrackingPlanConfig.TrackingPlan.Version
+			rudderTyperTPID := misc.MapLookup(singularEvent, "context", "ruddertyper", "trackingPlanId")
+			rudderTyperTPVersion := misc.MapLookup(singularEvent, "context", "ruddertyper", "trackingPlanVersion")
+			if rudderTyperTPID != nil && rudderTyperTPVersion != nil {
+				trackingPlanId = rudderTyperTPID.(string)
+				trackingPlanVersion = rudderTyperTPVersion.(int)
```

Comment:
We should check for the type conversion as well.
```suggestion
				if id, ok := rudderTyperTPID.(string); ok {
					trackingPlanId = id
				}
				if version, ok := rudderTyperTPVersion.(int); ok {
					trackingPlanVersion = version
				}
```